infra/aws/terraform/prow-build-cluster/README.md

```diff
@@ -0,0 +1,69 @@
+# Provisioninig EKS clusters
+
+## Prod vs Canary
+
+These scripts support provisioning two types of EKS clusters. One is meant for hosting prow jobs
+on production and the other one is for testing infrastructure changes before promoting them to
+production.
+
+Here are some differences between canary and production setups:
+* cluster name,
+* cluster admin IAM role name,
+* secrets-manager IAM policy name,
+* canary is missing k8s prow OIDC provider and corresponding role,
+* subnet setup is different,
+* instance type and autoscaling paramethers (mainly for saving),
+
+## Provisioning Cluster
+
+Running installation from scratch is different than consecutive invocations of Terraform.
+First run creates a role that can be later assumed by other users. Becasue of that additional
+variable has to be set:
+
+```bash
+# For provisioning Prod:
+export PROW_CLUSTER=prod
+# For provisioning Canary:
+export PROW_CLUSTER=canary
+
+# Just making sure we don't have state cached locally.
+make clean
+
+ASSUME_ROLE=false make init
+ASSUME_ROLE=false make apply
+```
+
+Once the infrastructure is provisioned, next step is RBAC setup:
+
+```bash
+# Fetch & update kubeconfig.
+# For Prod:
+aws eks update-kubeconfig --region us-east-2 --name prow-build-cluster
+# For Canary:
+aws eks update-kubeconfig --region us-east-2 --name prow-build-canary-cluster
```

Comment:
Isn't it required to change the generated kubeconfig to add the role ARN?

Reply:
This has to be done once, only after provisioning. I assume here that credentials used for updating kubeconfig were used for cluster provisioning. In such case you don't need to assume role. In fact, you need to create RBACs first in order to assign cluster-admin permissions to that assumed role.

Reply:
But your point it good. After cluster is installed, assuming role is required. I think I'll add readme section about connecting to cluster.